pkg/client/client.go

```diff
@@ -13,3 +20,34 @@ type VMClient interface {
 	StartVM(id string) error
 	Close() error
 }
+
+// SourceClientFactory provides default client factory implementation
+type SourceClientFactory struct{}
+
+// NewSourceClientFactory creates new factory
+func NewSourceClientFactory() *SourceClientFactory {
+	return &SourceClientFactory{}
+}
+
+// NewOvirtClient creates new Ovirt clients
+func (f *SourceClientFactory) NewOvirtClient(dataMap map[string]string) (VMClient, error) {
+	return ovirtclient.NewRichOvirtClient(&ovirtclient.ConnectionSettings{
+		URL:      dataMap["apiUrl"],
+		Username: dataMap["username"],
+		Password: dataMap["password"],
+		CACert:   []byte(dataMap["caCert"]),
+	})
+}
+
+// NewVmwareClient creates new VMWare clients
+func (f *SourceClientFactory) NewVmwareClient(dataMap map[string]string) (VMClient, error) {
+	insecure, err := strconv.ParseBool(dataMap["insecure"])
```

Comment:
Can you add support for the secure method? And drop the insecure?

Reply:
I'm not sure I understand the question. The client is secure by default and only is insecure if user specifies `insecure=true` in order to disable ssl verification. I'll make sure I handle the case where `insecure` isn't present in the dataMap at all to make sure I've got it defaulting to secure.

Reply:
Yes, I've meant add support to CA cert, and don't use insecure at all actually. We don't support `insecure` for oVirt as well.

Reply:
Ah, understood. Can do.